### ISSUE TITLE
In lua, lua51, lua53, luajit fix broken mintty patch and remove winpty wrappers and dependencies

### ISSUE THREAD

**Issue by FractalU** [msys2/MINGW-packages#17472]
### Description / Steps to reproduce the issue

There are winpty wrappers in luajit and lua53 but not in lua and lua51. Additionally, these wrappers cause problems with stdin, stdout, stderr redirection. E.g. if you type `luajit -e "print([[test]])" | cat` you get the following error.
```
stdout is not a tty
```
Expected output
```
test
```
If you type `echo "print([[test]])" | luajit` instead you get
```
stdin is not a tty
```
Expected output
```
test
```
BTW. in lua the `[[...]]` syntax represents a multi-line string, it's useful for terminal commands to avoid conflicts with '' or "".

This problem is already mentioned in the issue https://github.com/msys2/MINGW-packages/issues/949. 

In lua and lua51 this issue has been addressed in the commit https://github.com/msys2/MINGW-packages/commit/dfc42e74549a10ca69bb352415e1f7033a8ec3a9. However, the containing fix is broken in many ways. The corresponding patch seemingly works because it applys that fix in the wrong position of the C preprocessor and therefore the code containing it will never get compiled in. In lua the 'LUA_USE_POSIX' and in lua51 the 'LUA_USE_ISATTY' macros aren't defined, but the patch is putting the fix in the conditions assuming they are defined. Essentially, the patch does nothing at all. The code of that fix contains a function call whose name is mistyped. The proper name is called 'is_cygpty', not 'iscygpty'. Furthermore, the invocation of 'is_cygpty' alone is wrong. It needs to be called together with 'isatty'. The correct code looks like `isatty(...) || is_cygpty(...)`. The patch for the python package in [/mingw-w64-python/0059-cygpty-isatty.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-python/0059-cygpty-isatty.patch) contains exactly that. At last, a linker error occurs if lua is being compiled to a dynamically linked library, which is the case here.
```
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe:
 lua.o:lua.c:(.text+0x1017): undefined reference to `is_cygpty'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [Makefile:65: lua.exe] Error 1
```
This is because none of the functions in the `iscygpty.c` file is being marked as `__declspec(dllexport)` and therefore cannot be linked dynamically. Note that all lua functions are being marked as `LUA_API` which implicitly has `__declspec(dllexport)`.

The patch files should be renamed from `0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch` to something like `0001-When-running-lua-on-mintty-or-other.patch`. From mintty it's clear that lua is running as a WIN32 program. Also, the luac program has nothing to do with that. It's the lua.exe that needs patching, not luac.exe. It can also be renamed to `0001-mintty-cygpty-isatty.patch` similar to the corresponding patch in the python package.

In lua, lua51, lua53 and luajit packages I modified the patch files and built with them. Below is a summary of all the changes I made.

### In lua [0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua/0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch)

Change from
```
diff --git a/src/Makefile b/src/Makefile
index a99735a..b495b55 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -33,7 +33,7 @@ CMCFLAGS=
 PLATS= guess aix bsd c89 freebsd generic ios linux linux-readline macosx mingw posix solaris
 
 LUA_A=	liblua.a
-CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o
+CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o iscygpty.o
 LIB_O=	lauxlib.o lbaselib.o lcorolib.o ldblib.o liolib.o lmathlib.o loadlib.o loslib.o lstrlib.o ltablib.o lutf8lib.o linit.o
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
 
diff --git a/src/iscygpty.c b/src/iscygpty.c
```
To
```
diff --git a/src/Makefile b/src/Makefile
index a99735a..b495b55 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -46,1 +46,1 @@
-ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
+ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O) iscygpty.o
@@ -63,6 +63,6 @@
-$(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+$(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
+	$(CC) -o $@ $(LDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
 	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
@@ -160,2 +160,3 @@
 
+iscygpty.o: iscygpty.c iscygpty.h
 lapi.o: lapi.c lprefix.h lua.h luaconf.h lapi.h llimits.h lstate.h \
@@ -212,1 +212,1 @@
-lua.o: lua.c lprefix.h lua.h luaconf.h lauxlib.h lualib.h
+lua.o: lua.c lprefix.h lua.h luaconf.h lauxlib.h lualib.h iscygpty.h
diff --git a/src/iscygpty.c b/src/iscygpty.c
```
And from
```
diff --git a/src/lua.c b/src/lua.c
index b5b884b..3eb2c33 100644
--- a/src/lua.c
+++ b/src/lua.c
@@ -20,6 +20,7 @@
 #include "lauxlib.h"
 #include "lualib.h"
 
+#include "iscygpty.h"
 
 #if !defined(LUA_PROGNAME)
 #define LUA_PROGNAME		"lua"
@@ -402,7 +403,11 @@ static int handle_luainit (lua_State *L) {
 #if defined(LUA_USE_POSIX)	/* { */
 
 #include <unistd.h>
+#ifdef _WIN32
+#define lua_stdin_is_tty()	iscygpty(0)
+#else
 #define lua_stdin_is_tty()	isatty(0)
+#endif
 
 #elif defined(LUA_USE_WINDOWS)	/* }{ */
 
-- 
2.30.0.windows.2
```
To
```
diff --git a/src/lua.c b/src/lua.c
index b5b884b..3eb2c33 100644
--- a/src/lua.c
+++ b/src/lua.c
@@ -20,6 +20,7 @@
 #include "lauxlib.h"
 #include "lualib.h"
 
+#include "iscygpty.h"
 
 #if !defined(LUA_PROGNAME)
 #define LUA_PROGNAME		"lua"
@@ -402,14 +402,14 @@
 #if defined(LUA_USE_POSIX)	/* { */
 
 #include <unistd.h>
-#define lua_stdin_is_tty()	isatty(0)
+#define lua_stdin_is_tty()	isatty(0) || is_cygpty(0)
 
 #elif defined(LUA_USE_WINDOWS)	/* }{ */
 
 #include <io.h>
 #include <windows.h>
 
-#define lua_stdin_is_tty()	_isatty(_fileno(stdin))
+#define lua_stdin_is_tty()	_isatty(_fileno(stdin)) || is_cygpty(_fileno(stdin))
 
 #else				/* }{ */
 
2.30.0.windows.2
```
### In lua [link-implib.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua/link-implib.patch)

Change from
```
 LUA_A=	liblua.a
+LUA_LA=	liblua.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o iscygpty.o
 LIB_O=	lauxlib.o lbaselib.o lcorolib.o ldblib.o liolib.o lmathlib.o loadlib.o loslib.o lstrlib.o ltablib.o lutf8lib.o linit.o
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
@@ -61,10 +62,10 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
```
To
```
 LUA_A=	liblua.a
+LUA_LA=	liblua.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o
 LIB_O=	lauxlib.o lbaselib.o lcorolib.o ldblib.o liolib.o lmathlib.o loadlib.o loslib.o lstrlib.o ltablib.o lutf8lib.o linit.o
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
@@ -61,10 +62,10 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUA_O) iscygpty.o $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
```
### In lua51 [0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua51/0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch)

Change from
```
diff --git a/src/Makefile b/src/Makefile
index a99735a..b495b55 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -25,7 +25,7 @@
 LUA_A=	liblua5.1.a
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
-	lundump.o lvm.o lzio.o
+	lundump.o lvm.o lzio.o iscygpty.o
 LIB_O=	lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o \
 	lstrlib.o loadlib.o linit.o
 
diff --git a/src/iscygpty.c b/src/iscygpty.c
```
To
```
diff --git a/src/Makefile b/src/Makefile
index a99735a..b495b55 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -38,1 +38,1 @@
-ALL_O= $(CORE_O) $(LIB_O) $(LUA_O) $(LUAC_O)
+ALL_O= $(CORE_O) $(LIB_O) $(LUA_O) $(LUAC_O) iscygpty.o
@@ -54,6 +54,6 @@
-$(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+$(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
 	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
@@ -122,2 +122,3 @@
 
+iscygpty.o: iscygpty.c iscygpty.h
 lapi.o: lapi.c lua.h luaconf.h lapi.h lobject.h llimits.h ldebug.h \
@@ -169,1 +169,1 @@
-lua.o: lua.c lua.h luaconf.h lauxlib.h lualib.h
+lua.o: lua.c lua.h luaconf.h lauxlib.h lualib.h iscygpty.h
diff --git a/src/iscygpty.c b/src/iscygpty.c
```
And from
```
diff --git a/src/luaconf.h b/src/luaconf.h
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -226,7 +226,12 @@
 */
 #if defined(LUA_USE_ISATTY)
 #include <unistd.h>
+#ifdef _WIN32
+#include "iscygpty.h"
+#define lua_stdin_is_tty()	iscygpty(0)
+#else
 #define lua_stdin_is_tty()	isatty(0)
+#endif
 #elif defined(LUA_WIN)
 #include <io.h>
 #include <stdio.h>

-- 
2.30.0.windows.2
```
To
```
diff --git a/src/luaconf.h b/src/luaconf.h
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -227,8 +227,10 @@
 #if defined(LUA_USE_ISATTY)
 #include <unistd.h>
+#include "iscygpty.h"
-#define lua_stdin_is_tty()	isatty(0)
+#define lua_stdin_is_tty()	isatty(0) || is_cygpty(0)
 #elif defined(LUA_WIN)
 #include <io.h>
 #include <stdio.h>
+#include "iscygpty.h"
-#define lua_stdin_is_tty()	_isatty(_fileno(stdin))
+#define lua_stdin_is_tty()	_isatty(_fileno(stdin)) || is_cygpty(_fileno(stdin))
 #else

-- 
2.30.0.windows.2
```
### In lua51 [link-implib.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua51/link-implib.patch)

Change from
```
 LUA_A=	liblua5.1.a
+LUA_LA=	liblua5.1.a
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
 	lundump.o lvm.o lzio.o iscygpty.o
@@ -52,10 +53,10 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
```
To
```
 LUA_A=	liblua5.1.a
+LUA_LA=	liblua5.1.a
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
 	lundump.o lvm.o lzio.o
@@ -52,10 +53,10 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) iscygpty.o $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
```
### In lua53 [implib.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua53/implib.patch)

Change from
```
@@ -114,7 +114,7 @@
 
 mingw:
 	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua5.3.exe" \
-	"AR=$(CC) -shared -o" "RANLIB=strip --strip-unneeded" \
+	"AR=$(CC) -shared -Wl,--out-implib=liblua5.3.dll.a -o" "RANLIB=strip --strip-unneeded" \
 	"SYSCFLAGS=-DLUA_BUILD_AS_DLL" "SYSLIBS=" "SYSLDFLAGS=-s" lua5.3.exe
 	$(MAKE) "LUAC_T=luac5.3.exe" luac.exe
 
```
To
```
@@ -31,1 +31,2 @@
 LUA_A=	liblua5.3.a
+LUA_LA=	liblua5.3.a
@@ -60,8 +61,8 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUA_O) iscygpty.o $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
 
@@ -114,7 +114,7 @@
 
 mingw:
-	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua5.3.exe" \
+	$(MAKE) "LUA_A=lua53.dll" "LUA_LA=liblua5.3.dll.a" "LUA_T=lua5.3.exe" \
-	"AR=$(CC) -shared -o" "RANLIB=strip --strip-unneeded" \
+	"AR=$(CC) -shared -Wl,--out-implib=liblua5.3.dll.a -o" "RANLIB=strip --strip-unneeded" \
 	"SYSCFLAGS=-DLUA_BUILD_AS_DLL" "SYSLIBS=" "SYSLDFLAGS=-s" lua5.3.exe
 	$(MAKE) "LUAC_T=luac5.3.exe" luac.exe
 
```
### In lua [PKGBUILD](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua/PKGBUILD)

Remove
```
  # Check we do not encounter:
  # "stdin is not a tty"
  set -x
  echo "print(\"Hello World\")" > hello-world.lua
  ./src/luac.exe - < hello-world.lua
  if [[ ! -f luac.out ]]; then
    echo "ERROR :: luac.exe stdin read failed to create luac.out"
    return 1
  fi
  rm hello-world.lua
  rm luac.out
```
### In lua51 [PKGBUILD](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua51/PKGBUILD)

Remove
```
  # Check we do not encounter:
  # "stdin is not a tty"
  set -x
  echo "print(\"Hello World\")" > hello-world.lua
  ./src/luac5.1.exe - < hello-world.lua
  if [[ ! -f luac.out ]]; then
    echo "ERROR :: luac5.1.exe stdin read failed to create luac.out"
    return 1
  fi
  rm hello-world.lua
  rm luac.out
```
### In lua53 [PKGBUILD](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-lua53/PKGBUILD)

Remove
```
  # Use winpty to wrap the exe when executed from bash. Please don't move this into a patch as hopefully one day we won't need this hack.
  local _exename="lua5.3"
  local _execname="luac5.3"
  #handle lua.exe and make lua script and symlink equivilant
  mv "${pkgdir}"${MINGW_PREFIX}/bin/${_exename}.exe "${pkgdir}"${MINGW_PREFIX}/bin/${_exename}_exe
  echo "#!/usr/bin/env bash" > "${pkgdir}${MINGW_PREFIX}/bin/${_exename}"
  echo '/usr/bin/winpty "$( dirname ${BASH_SOURCE[0]} )/'${_exename}'.exe" "$@"' >> "${pkgdir}${MINGW_PREFIX}/bin/${_exename}"
  mv "${pkgdir}"${MINGW_PREFIX}/bin/${_exename}_exe "${pkgdir}"${MINGW_PREFIX}/bin/${_exename}.exe

  #handle luac.exe and make luac script and symlink equivilant
  mv "${pkgdir}"${MINGW_PREFIX}/bin/${_execname}.exe "${pkgdir}"${MINGW_PREFIX}/bin/${_execname}_exe
  echo "#!/usr/bin/env bash" > "${pkgdir}${MINGW_PREFIX}/bin/${_execname}"
  echo '/usr/bin/winpty "$( dirname ${BASH_SOURCE[0]} )/'${_execname}'.exe" "$@"' >> "${pkgdir}${MINGW_PREFIX}/bin/${_execname}"
  mv "${pkgdir}"${MINGW_PREFIX}/bin/${_execname}_exe "${pkgdir}"${MINGW_PREFIX}/bin/${_execname}.exe
```
### In luajit [PKGBUILD](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-luajit/PKGBUILD)

Remove
```
  echo "#!/usr/bin/env bash" > "${pkgdir}${MINGW_PREFIX}/bin/${_realname}"
  echo '/usr/bin/winpty "$( dirname ${BASH_SOURCE[0]} )/'${_realname}'.exe" "$@"' >> "${pkgdir}${MINGW_PREFIX}/bin/${_realname}"
```
When removing those wrappers also remove the corresponding winpty dependency. The lua51 package has the winpty dependency but no wrappers for it.

Here are the links to the modified patch files I used. Note that the patches are hand edited. It could be the case that some adjustments are required.

### Expected behavior

- lua and lua51 have the broken `0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch` patch fixed and renamed to something like `0001-mintty-cygpty-isatty.patch`.
- luajit and lua53 have the `001-mintty-cygpty-isatty.patch` added.
- luajit and lua53 have the winpty wrappers removed.
- luajit, lua51 and lua53 have the winpty dependency removed.

Basically all the lua packages have the same mintty fix as python and all the winpty stuff removed.

### Actual behavior

- In lua and lua51 the `0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch` patch is broken.
- In luajit and lua53 the `0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch` patch is missing.
- luajit and lua53 have winpty wrappers.
- luajit, lua51 and lua53 depends on winpty.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22621

### MINGW environments affected

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANG32
- [X] CLANGARM64

### Are you willing to submit a PR?

No

**Comment by revelator** [comment]
nice catch :)

**Comment by FractalU** [comment]
Yeah, it's quite a huge worm I caught. Though, this issue is the last one for lua I know. After this fix all the mentioned lua packages should just work. Also, I know that luarocks has issues. I'm currently looking into it. Additionally, a similar mintty patch can be applied to the libuv library, by which nodejs depends on. The winpty wrapper for nodejs can then be removed. Unfortunately, it's not that simple, because libuv has it's own way of handling the tty stuff. Therefore, the code for that patch looks differently.

**Comment by revelator** [comment]
are you able to push it yourself ? else we have to hope one of the devs drops by here.

**Comment by revelator** [comment]
nodejs has been a bit wonky since we got it back among the packages, i rewrote a rather large part of the internal paths in my personal build (mostly problems with the use of the windows ~-dp0 variable but also a few that involved winpty) it has worked here since but it would be nice to avoid atleast parts of the hackery.

**Comment by revelator** [comment]
great :)

**Comment by FractalU** [comment]
I've added a pull request. This issue is related to https://github.com/msys2/MINGW-packages/issues/949. So that one can be closed as well if this one is closed.
